app/src/main/java/com/jetbrains/handson/mpp/mobile/MainActivity.kt

```diff
@@ -1,20 +1,95 @@
 package com.jetbrains.handson.mpp.mobile
 
+
 import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Spinner
 import android.widget.TextView
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.jetbrains.handson.mpp.mobile.models.OriginStation
+import com.jetbrains.handson.mpp.mobile.models.OutboundJourneys
+
+class RecycleAdapter(private val journeyList: ArrayList<OutboundJourneys>) : RecyclerView.Adapter<RecycleAdapter.ViewHolder>() {
+    // holder class to hold reference
+    inner class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
+        //get view reference
+        var stationName: TextView = view.findViewById(R.id.station_name) as TextView
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
+        // create view holder to hold reference
+        return ViewHolder( LayoutInflater.from(parent.context).inflate(R.layout.list_item, parent, false))
+    }
+
+    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
+        //set values
+        val journey = journeyList[position]
+        holder.stationName.text =  journey.originStation.displayName + " > " + journey.destinationStation.displayName + " At " + journey.departureTime.split("T")[1].split("+")[0].split(".")[0]
```

Comment:
same here re variable names might be clearer - what are you doing with the time? also there might be a datetime library you could use to format the string as you want it. and maybe string interpolation rather than str + str would look neater?